pkg/cmd/util/clientcmd/factory.go

```diff
@@ -45,12 +47,58 @@ func New(flags *pflag.FlagSet) *Factory {
 
 	// TODO: there should be two client configs, one for OpenShift, and one for Kubernetes
 	clientConfig := DefaultClientConfig(flags)
+	clientConfig = defaultingClientConfig{clientConfig}
 	f := NewFactory(clientConfig)
 	f.BindFlags(flags)
 
 	return f
 }
 
+// defaultingClientConfig detects whether the provided config is the default, and if
+// so returns an error that indicates the user should set up their config.
+type defaultingClientConfig struct {
+	nested kclientcmd.ClientConfig
+}
+
+// RawConfig calls the nested method
+func (c defaultingClientConfig) RawConfig() (kclientcmdapi.Config, error) {
```

Comment:
This is a wrapper... so we are implementing the interface.

On Thu, Nov 5, 2015 at 11:02 AM, David Eads <notifications@github.com>
wrote:

> In pkg/cmd/util/clientcmd/factory.go
> <https://github.com/openshift/origin/pull/5722#discussion_r44029086>:
>
> >  	f := NewFactory(clientConfig)
> >  	f.BindFlags(flags)
> >
> >  	return f
> >  }
> >
> > +// defaultingClientConfig detects whether the provided config is the default, and if
> > +// so returns an error that indicates the user should set up their config.
> > +type defaultingClientConfig struct {
> > +	nested kclientcmd.ClientConfig
> > +}
> > +
> > +// RawConfig calls the nested method
> > +func (c defaultingClientConfig) RawConfig() (kclientcmdapi.Config, error) {
>
> I'm used to method receivers being *struct. Any reason to avoid that here?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/5722/files#r44029086>.
>